Jobs.Fetcher.TikTok/Helpers/DatabaseManager.cs

```diff
@@ -0,0 +1,129 @@
+                ");
+                cmd.Parameters.AddWithValue("account_name", account_name);
+                TikTokUsers tiktokUser = null;
+                using (var reader = cmd.ExecuteReader()) {
+                    if (reader.Read()) {
+                        tiktokUser = new TikTokUsers(){
+                            //UserId = reader.GetString(0),
+                            Name = reader.GetString(1)
+                        };
+                    }
+                }
+                return tiktokUser;
+            }
+        }
+
+        public static string GetTikTokId(string username, DataLakeTikTokContext dbContext) {
+            var now = DateTime.UtcNow;
+            var valuetobereturned = dbContext.Authors.Where(m => m.UniqueId == username)
+                                        .Select(m => m.Id)
+                                        .FirstOrDefault();
+            return valuetobereturned;
+        }
+
+        public static bool TikTokUserExists(string username) {
+            using (var connection = new NpgsqlConnection(ConnectionString()))
+                using (var cmd = connection.CreateCommand()) {
+                    connection.Open();
+                    cmd.CommandText = String.Format(@"
+                        SELECT
+                            COUNT(*)
+                        FROM
+                            account_name
+                        WHERE
+                            account_name = @username
+                        ");
+                    cmd.Parameters.AddWithValue("username", username);
+                    using (var reader = cmd.ExecuteReader()) {
+                        if (reader.Read()) {
+                            return reader.GetDecimal(0) > 0;
+                        }
+                    }
+                    return false;
+                }
+        }
+
+        public static bool TikTokScraperTablesExist() {
+            using (var connection = new NpgsqlConnection(ConnectionString()))
+                using (var cmd = connection.CreateCommand()) {
+                    connection.Open();
+                    cmd.CommandText = String.Format(@"
+                        SELECT EXISTS (
+                            SELECT FROM information_schema.tables 
+                            WHERE table_name   = 'video_info'
+                        );");
+                    using (var reader = cmd.ExecuteReader()) {
+                        if (reader.Read()) {
+                            var returnValue = reader.GetBoolean(0);
+                            return returnValue;
+                        }
+                    }
+                    return false;
+                }
+        }
+
+        public static DateTime GetLastFetch(string authorId, DataLakeTikTokContext dbContext) {
+            var now = DateTime.UtcNow;
+            return dbContext.AuthorStats.Where(m => m.AuthorId == authorId && m.ValidityStart <= now && m.ValidityEnd > now)
```

Comment:
I would also like to establish a standard for this timestamp range check. Are we going to check `ValidityEnd > now` or `ValidityEnd == DateTime.MaxValue`? I think the first one is the one normally used in the codebase

Reply:
Yeah, I think ```ValidityEnd > now``` is used more often, so I'll check any places where I might have used something different and change it